src/core/cgroup.c

```diff
@@ -848,10 +851,12 @@ static void cgroup_context_apply(Unit *u, CGroupMask mask, ManagerState state) {
         if ((mask & CGROUP_MASK_MEMORY) && !is_root) {
                 if (cg_all_unified() > 0) {
                         uint64_t max = c->memory_max;
+                        uint64_t swap_max = c->memory_swap_max;
 
-                        if (cgroup_context_has_unified_memory_config(c))
+                        if (cgroup_context_has_unified_memory_config(c)) {
                                 max = c->memory_max;
-                        else {
+                                swap_max = c->memory_swap_max;
```

Comment:
Heh, the logic is kinda weird here. Also with @max. How about removing initialization of @max and init @swap_max to CGROUP_LIMIT_MAX? Or just init both @max and @swap_max to CGROUP_LIMIT_MAX.